libs/ngxtension/trackby-id-prop/src/trackby-id-prop.ts

```diff
@@ -0,0 +1,42 @@
+//INSPIRED BY https://medium.com/ngconf/make-trackby-easy-to-use-a3dd5f1f733b
+import { NgForOf } from '@angular/common';
+import {
+	Directive,
+	Input,
+	Provider,
+	inject,
+	type NgIterable,
+} from '@angular/core';
+
+@Directive({
+	selector: '[ngForTrackById]',
+	standalone: true,
+})
+export class NgForTrackByIdDirective<T extends { id: string | number }> {
+	@Input() ngForOf!: NgIterable<T>;
+	private ngFor = inject(NgForOf<T>, { self: true });
+
+	constructor() {
+		this.ngFor.ngForTrackBy = (index: number, item: T) => item.id;
+	}
+}
+
+@Directive({
+	selector: '[ngForTrackByProp]',
+	standalone: true,
+})
+export class NgForTrackByPropDirective<T> {
+	@Input() ngForOf!: NgIterable<T>;
+	private ngFor = inject(NgForOf<T>, { self: true });
+
+	@Input()
+	set ngForTrackByProp(trackByProp: keyof T) {
+		if (!trackByProp) return; //throw new Error("You must specify trackByProp:'VALID_PROP_NAME'");
+		this.ngFor.ngForTrackBy = (index: number, item: T) => item[trackByProp];
+	}
+}
+
+export const TrackByDirectives: Provider[] = [
```

Comment:
question: is `Provider[]` a correct type for this? These aren't Providers are they?

Reply:
Yeah you are right! Provider is not the right type, It works just becouse Provider include Type<T>
I checked the types in angular and it's `imports?: (Type<any> | ReadonlyArray<any>)[]`

Do you prefer that I explicit the type as :Type<any>[]  or simply remove the TS type declaration and simply `export const XXX = [D1, D2] as const;` //This will infer the ReadonlyArray or Tuple I think

Reply:
Let’s try as const